pkg/kubectl/cmd/delete.go

```diff
@@ -116,6 +116,7 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		SelectAllParam(deleteAll).
 		ResourceTypeOrNameArgs(false, args...).RequireObject(false).
 		Flatten().
+		Latest().
```

Comment:
If the rest handler makes a successful etcd update as part of fulfilling a delete-with-grace-period-0 API call, I would expect any odd logic around tolerating a not found on the subsequent etcd delete to be done inside the rest handler.